bindings/pyroot/pythonizations/python/ROOT/pythonization/_roofit/_rooabspdf.py

```diff
@@ -10,45 +10,34 @@
 # For the list of contributors see $ROOTSYS/README/CREDITS.                    #
 ################################################################################
 
-r"""
-/**
-\class RooAbsPdf
-\brief \parblock \endparblock
-\htmlonly
-<div class="pyrootbox">
-\endhtmlonly
-
-## PyROOT
-
-Some member functions of RooAbsPdf that take a RooCmdArg as argument also support keyword arguments.
-So far, this applies to RooAbsPdf::fitTo, RooAbsPdf::plotOn, RooAbsPdf::generate, RooAbsPdf::paramOn, RooAbsPdf::createCdf,
-RooAbsPdf::generateBinned, RooAbsPdf::createChi2, RooAbsPdf::prepareMultiGen and RooAbsPdf::createNLL.
-For example, the following code is equivalent in PyROOT:
-\code{.py}
-# Directly passing a RooCmdArg:
-pdf.fitTo(data, ROOT.RooFit.Range("r1"))
-
-# With keyword arguments:
-pdf.fitTo(data, Range="r1")
-\endcode
-
-\htmlonly
-</div>
-\endhtmlonly
-*/
-"""
 
 from ._rooabsreal import RooAbsReal
 from ._utils import _kwargs_to_roocmdargs
 
 
 class RooAbsPdf(RooAbsReal):
+
+    _doxygen = """Some member functions of RooAbsPdf that take a RooCmdArg as argument also support keyword arguments.
+    So far, this applies to RooAbsPdf::fitTo, RooAbsPdf::plotOn, RooAbsPdf::generate, RooAbsPdf::paramOn, RooAbsPdf::createCdf,
+    RooAbsPdf::generateBinned, RooAbsPdf::createChi2, RooAbsPdf::prepareMultiGen and RooAbsPdf::createNLL.
+    For example, the following code is equivalent in PyROOT:
+    \code{.py}
+    # Directly passing a RooCmdArg:
+    pdf.fitTo(data, ROOT.RooFit.Range("r1"))
+
+    # With keyword arguments:
+    pdf.fitTo(data, Range="r1")
+    \endcode"""
+
     def fitTo(self, *args, **kwargs):
         # Redefinition of `RooAbsPdf.fitTo` for keyword arguments.
         # The keywords must correspond to the CmdArg of the `fitTo` function.
         args, kwargs = _kwargs_to_roocmdargs(*args, **kwargs)
         return self._fitTo(*args, **kwargs)
 
+    fitTo._cpp_signature = "RooAbsPdf::fitTo(RooAbsData&, const RooCmdArg&, const RooCmdArg&, const RooCmdArg&, const RooCmdArg&, const RooCmdArg&, const RooCmdArg&, const RooCmdArg&, const RooCmdArg&)"
```

Comment:
This can also be a list of signatures, right?

How would you specify it in that case?

I would just decide on a format that you parse from `__doc__` (see my previous comment).

Reply:
Yes it can be. Usually it's not, because even if there are multiple overloads of a function, we usually add the documentation to only one of them in RooFit and put `\see` to the other ones. In any case, I added support for multiple signatures in the `print_roofit_pyz_doctrings.py ` script.

Reply:
Actually I'm not a bit fan of parsing the signatures from the docstring. If the information is already structured, why serialize it in a string to it has to be parsed back? Many errors can happen there.

As a compromise, I suggest now a `cpp_signature` decorator for syntactic sugar. Maybe you like it now? So a member function in the mirror class would look like:

```Python
    @cpp_signature("RooAbsPdf::fitTo()")
    def fitTo(self, *args, **kwargs):
        """This function is pythonized with the command argument pythonization.
        The keywords must correspond to the CmdArgs of the RooAbsPdf::fitTo() function.
        """
        ...
```

Reply:
That looks nice, `@cpp_signature` could also accept a list of signatures if needed instead of just one string.